hs/src/Reach/Optimize.hs

```diff
@@ -264,6 +264,10 @@ instance {-# OVERLAPPING #-} Optimize a => Optimize (DLinExportVal a) where
     DLEV_LArg at a  -> DLEV_LArg at <$> opt a
     DLEV_Fun at a b -> DLEV_Fun at a <$> opt b
 
+instance {-# OVERLAPPING #-} (Eq a, Sanitize a, Extract a) => Optimize (DLExportinBlock a) where
+  opt = \case
+    DLExportinBlock b r -> DLExportinBlock <$> opt b <*> opt r
```

Comment:
These are going to need new scopes so we don't rely on the evaluation order being anything in particular. If instead we went for there being one block, then it would not need to be separate scopes for each one